Bottom: 171d259731d98d261a2c94259057a57aeeb69854
Top:    3656d211d57cd5f9845c3beb77414d2ed3d25ec2
Author: Ian Kent <raven@themaw.net>
Date:   2013-05-02 15:51:27 +0800

Refresh of autofs-5.0.7-use-MS_UNBINDABLE-for-autofs-mounts.patch

---

diff --git a/include/automount.h b/include/automount.h
index e72fa0d..6b1af1b 100644
--- a/include/automount.h
+++ b/include/automount.h
@@ -75,6 +75,12 @@ int load_autofs4_module(void);
 #define SMB_SUPER_MAGIC    0x0000517BL
 #define CIFS_MAGIC_NUMBER  0xFF534D42L
 
+#ifndef MS_UNBINDABLE
+#define MS_UNBINDABLE (1 << 17)
+#endif
+
+unsigned long mountflags = MS_UNBINDABLE|MS_MGC_VAL;
+
 /* This sould be enough for at least 20 host aliases */
 #define HOST_ENT_BUF_SIZE	2048
 
diff --git a/lib/mounts.c b/lib/mounts.c
index cfd303c..d26209b 100644
--- a/lib/mounts.c
+++ b/lib/mounts.c
@@ -103,10 +103,12 @@ unsigned int query_kproto_ver(void)
 	}
 
 	if (mount("automount", t_dir, "autofs", MS_UNBINDABLE|MS_MGC_VAL, options)) {
-		close(pipefd[0]);
-		close(pipefd[1]);
-		rmdir(t_dir);
-		return 0;
+		if (mount("automount", t_dir, "autofs", MS_UNBINDABLE|MS_MGC_VAL, options)) {
+			close(pipefd[0]);
+			close(pipefd[1]);
+			rmdir(t_dir);
+			return 0;
+		}
 	}
 
 	close(pipefd[1]);
